### ISSUE TITLE
INPUT command not working

### ISSUE THREAD

**Issue by bwganblack** [boriel/zxbasic#571]
The INPUT command generates a flashing cursor but is otherwise unresponsive. Tested with the example file:

```
#include <input.bas>

print at 10, 5; "Type something: ";
a$ = input(20)
print
print "You typed: "; a$
```
Using Nextbuildv7, ZXbasic ver  :  1.15.2

The problem appears to be line 42 of input.bas:  `DO LOOP UNTIL LastK <> 0`
The program gets stuck in this loop and pressing a key doesn't exit it, proven by inserting a BEEP command into the loop.

Using labels and GOTO commands to replace both the DO...LOOPS in the INPUT function, I was able to create a ersatz version which works. Not sure if this is a bug or if I'm missing something obvious

Thanks.

**Comment by boriel** [comment]
Are you compiling using -O4?
This optimization is too aggressive for this routine (LastK is a _Volatile_ variable and it's not supported yet). If so, compiling with -O3 should work.

**Comment by boriel** [comment]
Update: This function was successfully tested on ZX 128k. I'll test on the ZX Next and let you know.

**Comment by bwganblack** [comment]
I'm compiling through Nextbuild, presumably using the default optimization level as I haven't even looked into how I would change this yet. I'm just using emulators: I've generated a NEX file and run it on CSpect and I've generated a TAP file and run it on Zero. Same result, unresponsive.

PS: I'd love to participate on the forum but I'm getting an ERROR 403 - Forbidden! message when I try to connect to any page. Perhaps my IP address is blocked? Apologies if I appeared to be spamming. I wasn't receiving the activation email and thought I might have filled in my details wrong. Is this sent out automatically or manually?  Perhaps I just needed to be more patient!

**Comment by bwganblack** [comment]
Coming back to this, I found how to change the optimization level when compiling with NextBuild/VS Code by editing the nextbuild.py file. You were quite correct @boriel, changing from level **4** (the default) to level **3** allows the above code to work.

I tested by generating a .tap file and running using Zero v0.7.1. By adding `#include <nextlib.bas>` it will also work when generating a .nex file and running using CSpect v2.15.1 (but not without).

Tested with ZXBasic v.1.16.4 and NextBuild v.7.3.1.

Thanks.
